### PR TITLE
Added new page for URL library

### PR DESCRIPTION
**Pull request by drpayyne**
## Purpose of this pull request

This pull request (PR) adds a new page `URL Library` to describe the utilities provided to manage URLs.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/url-library.html

## Links to Magento source code

-  https://github.com/magento/magento2/tree/2.4-develop/lib/internal/Magento/Framework/Url

whatsnew
Added a new topic describing the [URL Library](https://devdocs.magento.com/guides/v2.4/extension-dev-guide/framework/url-library.html).

### CODE REVIEW

File: src/guides/v2.3/extension-dev-guide/framework/url-library.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+|--- |--- |
+| `+` | `-` |
+| `/` | `_` |
+| `=` | `,` |
+
+### Decoder
+
+The [`Magento\Framework\Url\DecoderInterface`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Url/DecoderInterface.php){:target="_blank"} provides a method to `decode` the base64 encoded URL provided to it and also decodes the special characters described in the table below.
+
+|Special Character|Encoded Value|
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/url-library.md:28]: Shouldn't this say "Decoded Value" ?

File: src/guides/v2.3/extension-dev-guide/framework/url-library.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    DecoderInterface $decoder
+) {
+  $this->encoder = $encoder;
+  $this->decoder = $decoder;
+}
+
+/**
+ * Encodes URL to base64 format and escapes special characters
+ */
+public function encodeURL($url)
```
Comment (BarnyShergold) [inline src/guides/v2.3/extension-dev-guide/framework/url-library.md:64]: Should be a return type here and also DOCBLOCKs should have parameters & return types - applies to both methods.

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice new topic - just a few questions & suggestions

File: src/guides/v2.3/extension-dev-guide/framework/url-library.md (hunk 2 above)
Comment (drpayyne) [reply src/guides/v2.3/extension-dev-guide/framework/url-library.md:73]: Done, thank you!

File: src/guides/v2.3/extension-dev-guide/framework/url-library.md (hunk 1 above)
Comment (drpayyne) [reply src/guides/v2.3/extension-dev-guide/framework/url-library.md]: Done, thank you!

**Review by BarnyShergold — APPROVED** [review]
Nice work!

**Review by jeff-matthews — APPROVED** [review]
Nice work @drpayyne

**Review by jeff-matthews — CHANGES_REQUESTED** [review]
@drpayyne, please create a 2.4 symlink for the new topic.

**Comment by drpayyne** [comment]
Done, @jeff-matthews! Thank you.

**Review by jeff-matthews — APPROVED** [review]

**Comment by jeff-matthews** [comment]
running tests

**Comment by dshevtsov** [comment]
@jeff-matthews , the label for version was missing. I will add it.

**Comment by dshevtsov** [comment]
@drpayyne , your link to Magento source code points to the 2.4-develop branch. Does this mean that your contribution relates to the code in development rather than released code?

**Comment by drpayyne** [comment]
@dshevtsov No, it's available in the main releases as well.

**Comment by dshevtsov** [comment]
@drpayyne Could you please update it to https://github.com/magento/magento2/tree/2.4.1/lib/internal/Magento/Framework/Url to make it more specific for keeping in the GitHub history?

**Comment by drpayyne** [comment]
@dshevtsov Done, thank you!

**Pull request merged**